### PR TITLE
fix(cas): Load candidate streams in parallel batches

### PR DESCRIPTION
**Pull request by stbrody**
It may be easier to review this PR by reviewing the two commits that go into it one at a time. The first refactors the code without any functionality change, the second changes the way we load streams to add parallelism.

### CODE REVIEW

**Comment by v-stickykeys** [comment]
This is small but I looked it up because I wasn't sure--technically this is not "parallel" but it is "concurrent" so they will still be started sequentially. Should not be a problem though

File: src/services/anchor-service.ts [hunk 1]
```diff
@@ -241,78 +243,107 @@ export default class AnchorService {
...
+    const candidates = []
+    let index = 0
+    while (index < requests.length && candidates.length < streamCountLimit) {
+      const batchSize = Math.min(BATCH_SIZE, streamCountLimit - candidates.length)
+      const batchRequests = requests.slice(index, Math.min(index + batchSize, requests.length))
 
-      let docId
-      try {
-        request = requests[index];
+      const batchCandidates = await Promise.all(batchRequests.map((request) => { return this._loadCandidateForRequest(request) }))
```
[the remark this answers is not in the document]
Comment (stbrody) [reply src/services/anchor-service.ts:277]: If one of the promises rejects?  I'm actually not sure. But there's a try/catch around the entire body of `_loadCandidateForRequest` so it really shouldn't ever reject

**Comment by stbrody** [comment]
> This is small but I looked it up because I wasn't sure--technically this is not "parallel" but it is "concurrent" so they will still be started sequentially. Should not be a problem though

If I'm understanding the distinction properly then the same could be said of literally any javascript code, no?

**Comment by v-stickykeys** [comment]
> If I'm understanding the distinction properly then the same could be said of literally any javascript code, no?

Yeah I think that's true, unless you are using a subprocess

**Review by v-stickykeys — APPROVED** [review]
Yay CAS improvements!

**Pull request merged**